packages/data/src/registry.js

```diff
@@ -215,6 +215,10 @@ export function createRegistry( storeConfigs = {}, parent = null ) {
 	 * @param {Object} store Store instance object (getSelectors, getActions, subscribe).
 	 */
 	function registerStoreInstance( name, store ) {
+		if ( stores[ name ] ) {
+			throw new Error( 'duplicate store oh no' );
```

Comment:
It's going to be replaced by just `return;` eventually. The error is intentionally bad so that we don't forget about that 🙂

Reply:
Sounds good.

We could still do a `console.error()` though. I think it's something worth informing developers about.

Reply:
> We could still do a `console.error()` though.

That's a good idea, I added a `console.error` + `return` instead of throwing. It turns out that `registerBlockType` also does that.

On the other hand, `registerPlugin` prints a console error that plugin is already registered, but then proceeds to register the new version. Not sure if we really wanted that.

Reply:
Here is an opportunity to define and align the behavior for all those APIs. I saw also @noahtallen commenting in https://github.com/WordPress/gutenberg/pull/49134#discussion_r1140688858 that we could use `warning` from the `@wordpress/warning` package.

Reply:
Yes, I like the idea of using `warning` in these cases. Let's do that in a separate PR. I'd rather keep this one focused on the data package and its store registration code.